### PR TITLE
added hierarchical LDA classification scripts

### PR DESCRIPTION
**Pull request by MihailBogojeski**
This pull request was created due to the suggestion to add the hierarchical LDA classifier scripts to the main toolbox.

### CODE REVIEW

File: classification/apply_HLDA.m [hunk 1]
```diff
@@ -0,0 +1,61 @@
+function out = apply_HLDA(epo, ival, C)
+% TRAIN_LDA - linear discriminant analysis
```
Comment (BenjaminBlankertz) [inline classification/apply_HLDA.m]: % TRAIN_HLDA - Hierarchical Linear Discriminant Analysis

File: classification/apply_HLDA.m [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+misc_checkType(epo, 'STRUCT(x y t)');
+misc_checkType(ival, 'INT[2]');
+misc_checkType(C, 'STRUCT(seg final)');
+
+
+nSegments = length(C.seg);
+dims = size(epo.x);
+% convert interval values to indices
+ival_idx = find(epo.t == ival(1)) : find(epo.t == ival(2));
```
Comment (BenjaminBlankertz) [inline classification/apply_HLDA.m]: This is a bit problematic. epo.t might not include the exact values given (think, e.g., about the case of a sampling rate of 256Hz). Therefore you should do something like:
     ibeg= max([1 find(epo.t<=ival(1))]);
     iend= min([find(epo.t>=ival(2)) length(epo.t)]);
     ival_idx= ibeg:iend;
However, you could also use the function proc_selectIval.

File: classification/train_HLDA.m [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+%   APPLY_HLDA
+
+misc_checkType(epo, 'STRUCT(x y t)');
+misc_checkType(ival, 'INT[2]');
+misc_checkType(nSegments, 'INT');
+
+if isempty(ival)
+    ival = [epo.t(1), epo.t(end)];
+end
+% convert interval values to indices
```
Comment (BenjaminBlankertz) [inline classification/train_HLDA.m]: Use the function proc_selectIval.

File: classification/apply_HLDA.m (hunk 1 above)
Comment (BenjaminBlankertz) [inline classification/apply_HLDA.m]: % TRAIN_HLDA - Hierarchical Linear Discriminant Analysis

File: classification/apply_HLDA.m [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+%                           contains the individual segment classifiers and 
+%                           the top-level classifier. Must include the
+%                           fields 'seg' and 'final'
+%Returns:
+%   out: FLOAT[]        - an array containing the classifier score for each 
+%                           sample
+%Description:
+%   APPLY_HLDA applies a hierarchical LDA classifier given an epo structure,
+%               a training interval and a number of segments
+%
```
Comment (BenjaminBlankertz) [inline classification/apply_HLDA.m:25]: Add reference to the paper describing HLDA.

File: classification/apply_HLDA.m [hunk 5]
```diff
@@ -0,0 +1,61 @@
+function out = apply_HLDA(epo, ival, C)
```
Comment (BenjaminBlankertz) [inline classification/apply_HLDA.m]: If the apply function has arguments other than (fv, C) -- such as 'ival' here -- they cannot be used in crossvalidation. We should discuss, whether it makes sense to change this here.

Comment (MIkioBo) [inline classification/apply_HLDA.m]: We could get rid of the ival altogether (both in the train and apply functions), and simply pass the final X (with dimensions (time * channels) X samples) and the labels Y (similar to train_LDA in the toolbox). In this case, the needed interval should already be selected before passing X to the training function. In this way, the arguments and the  epo.t problems should be solved.

**Comment by MihailBogojeski** [comment]
We could probably leave out the parameter ival altogether, and build the signature in a similar way to the regular train_LDA function, i.e., we would require X ( with dimensions (time \* channels) X samples) and Y (the labels) to be passed directly. So the selection of the relevant interval should already be done before X is passed. In this way, we avoid both the problem with the crossvalidation and the selection of the interval using epo.t.
EDIT: It is better for X to have three dimensions (time X channels X samples), because otherwise the segmentation may separate two groups of channels for the same time point in different segments.

**Pull request merged**